SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -3122,28 +3124,32 @@ void CEulerSolver::Upwind_Residual(CGeometry *geometry, CSolver **solver_contain
 
       for (iVar = 0; iVar < nPrimVarGrad; iVar++) {
 
+        const su2double V_ij = V_j[iVar] - V_i[iVar];
+
         su2double Project_Grad_i = 0.0;
         su2double Project_Grad_j = 0.0;
 
         for (iDim = 0; iDim < nDim; iDim++) {
           Project_Grad_i += Vector_ij[iDim]*Gradient_i[iVar][iDim];
-          Project_Grad_j -= Vector_ij[iDim]*Gradient_j[iVar][iDim];
+          Project_Grad_j += Vector_ij[iDim]*Gradient_j[iVar][iDim];
         }
 
         if (limiter) {
-          if (van_albada) {
-            su2double V_ij = V_j[iVar] - V_i[iVar];
-            Limiter_i[iVar] = V_ij*( 2.0*Project_Grad_i + V_ij) / (4*pow(Project_Grad_i, 2) + pow(V_ij, 2) + EPS);
-            Limiter_j[iVar] = V_ij*(-2.0*Project_Grad_j + V_ij) / (4*pow(Project_Grad_j, 2) + pow(V_ij, 2) + EPS);
+          switch(config->GetKind_SlopeLimit_Flow()) {
+            case VAN_ALBADA_EDGE:
+              Limiter_i[iVar] = LimiterHelpers::vanAlbadaFunction(Project_Grad_i, V_ij);
+              Limiter_j[iVar] = LimiterHelpers::vanAlbadaFunction(Project_Grad_j, V_ij);
+              break;
+            case PIPERNO:
+              Limiter_i[iVar] = LimiterHelpers::pipernoFunction(Project_Grad_i, V_ij);
+              Limiter_j[iVar] = LimiterHelpers::pipernoFunction(Project_Grad_j, V_ij);
+              break;
           }
-          Primitive_i[iVar] = V_i[iVar] + Limiter_i[iVar]*Project_Grad_i;
-          Primitive_j[iVar] = V_j[iVar] + Limiter_j[iVar]*Project_Grad_j;
+          Project_Grad_i = Limiter_i[iVar]*((1.0-Kappa)*Project_Grad_i + 0.5*Kappa*V_ij);
```

Comment:
Is it standard practice to apply the limiter to the sum of the projection with the central average?
I'm asking because the point limiters are based only on the projection and the average is guaranteed to be TVD.

Reply:
> Is it standard practice to apply the limiter to the sum of the projection with the central average?

From what I've seen, it is standard. The definitions I've seen for edge limiters have the sum of the projection and central difference baked into the limiter definition.

> I'm asking because the point limiters are based only on the projection and the average is guaranteed to be TVD.

Now I'm wondering whether the point limiters should be computed with the projection and delta, or the sum and delta. The Inria papers I linked above, and some others I've found like [this NASA/NIA paper](http://www.ossanworld.com/hiroakinishikawa/My_papers/white_nishikawa_baurle_aiaa2019-0127.pdf) seem to use the sum in their limiter functions.

Reply:
That sounds like a reasonable modification.